### DISCUSSION TITLE
Error while Running cypress latest 13.4.0 in CI with allure adapter

### DISCUSSION THREAD

**Discussion by AhmadfBaker** [mmisty/cypress-allure-adapter#65] [Q&A] [answered]
**Node version 20.9 in cypress docker image**
This option will not have an effect in Firefox. Tests that rely on web security being disa
| Your configFile threw an error from: cypress. config.js
| We stopped running your tests because your config file crashed.
I Error: error: 0308010C:digital envelope routines: : unsupported
at new Hash (node: internal/crypto/hash: 68:19)
at Object.createHash (node: crypto: 138:10)
at module. exports (/app/node_modules /webpack/lib/util/createHash.js:135:53)
at NormalModule._initBuildHash(/app/node_modules/webpack/lib/NormalModule-js:417:16)
at handleParseError (/ app/node_modules/webpack/Lib/NormalModule. js:471:10)
at /app/node_modules/webpack/lib/NormalModule. js: 503:5
at /app/node_modules/webpack/lib/NormalModule.js:358:12
at /app/node_modules/loader-runner/lib/LoaderRunner. js:373:3
at iterateNormalLoaders (/ app/node_modules/loader-runner/lib/LoaderRunner. js: 214:10)
at Array. <anonymous> (/ app/node_modules/loader-runner/lib/LoaderRunner. js:205:4)
at Storage. finished (/app/node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:55
at /app/node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:91:9
at /app/node_modules/graceful-fs/graceful-fs.js:123:16

**Comment by mmisty** [comment]
Hi @AhmadfBaker 
I don't see anything related to cypress-allure-adapter (stack trace doesn't have any clues)
And I didn't understand what happened. 
Could you please add reproducible steps, your config file, screenshots and more information if you believe that issue is within cypress-allure-adapter

**Reply by AhmadfBaker** [reply]
@Hi @mmisty 
I solved the problem above, But i have now a problem that the allure-results folder is empty.

`"allure:report": "allure generate allure-results-aggregated-files --clean -o allure-report",` 
`"allure:clear": "npm run allure:size;rm -rf allure-results/*;rm -rf allure-results-aggregated-files/*;rm -rf  allure-report/*; rm -rf cypress/screenshots/*; rm -rf  cypress/videos/*;npm run allure:size",`
`"cypress": "npm run allure:clear;npx cypress run --browser chrome --spec cypress/e2e-v2/dram/by-dram/dram-die-variant-functionality-spec.cy.ts --env allure=true;npm run allure:report"`

BTW, I used npm run cypress

**Comment by mmisty** [comment, marked as answer]
Hi @AhmadfBaker 
Could you please
1 verify that `allureResults` variable in cypress config is set to 'allure-results'
2 delete `allure-results` folder
3 run `npx cypress run --browser chrome --spec cypress/e2e-v2/dram/by-dram/dram-die-variant-functionality-spec.cy.ts --env allure=true`
4 check whether `allure-results` folder exist with .json files

If it doesn't exist could you please send: 
 - e2e.ts (suppport/index.ts) file
 - cypress.config.ts
 - logs of cypress run
 - contents of `cypress/e2e-v2/dram/by-dram/dram-die-variant-functionality-spec.cy.ts`

**Comment by AhmadfBaker** [comment]
Hi, @mmisty After i solved the problem for the allure results, in the CI I still have a problem running cypress 13.4.0 with allure in CI 

This option will not have an effect in Firefox. Tests that rely on web security being disa
| Your configFile threw an error from: cypress. config.js
| We stopped running your tests because your config file crashed.
I Error: error: 0308010C:digital envelope routines: : unsupported
at new Hash (node: internal/crypto/hash: 68:19)
at Object.createHash (node: crypto: 138:10)
at module. exports (/app/node_modules /webpack/lib/util/createHash.js:135:53)
at NormalModule._initBuildHash(/app/node_modules/webpack/lib/NormalModule-js:417:16)
at handleParseError (/ app/node_modules/webpack/Lib/NormalModule. js:471:10)
at /app/node_modules/webpack/lib/NormalModule. js: 503:5
at /app/node_modules/webpack/lib/NormalModule.js:358:12
at /app/node_modules/loader-runner/lib/LoaderRunner. js:373:3
at iterateNormalLoaders (/ app/node_modules/loader-runner/lib/LoaderRunner. js: 214:10)
at Array. (/ app/node_modules/loader-runner/lib/LoaderRunner. js:205:4)
at Storage. finished (/app/node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:55
at /app/node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:91:9
at /app/node_modules/graceful-fs/graceful-fs.js:123:16

**Reply by mmisty** [reply]
Hi @AhmadfBaker 
The stack trace that you've attached is not related to cypress-allure-adapter anyhow.

**Reply by mmisty** [reply]
https://stackoverflow.com/questions/69692842/error-message-error0308010cdigital-envelope-routinesunsupported

**Comment by AhmadfBaker** [comment]
Thanks @mmisty 
I solved the problem, and everything works fine now. 

But i noticed that generate allure report now takes 13 min in my project, while it was take 2 min only. Do you aware of that performance? or it's related enable all these features? 
            allureAttachRequests:true,
            allureLogCyCommands: true,
            allureShowDuplicateWarn:true,
            allureShowTagsInTitle:true,

**Reply by mmisty** [reply]
@AhmadfBaker 
I assume if you have many requests there would be a lot of attachments - so you can play with  'allureAttachRequests' and check if it gets better
- How many file do you have in allure-results folder?
- What OS do you use?

**Reply by AhmadfBaker** [reply]
@mmisty Yes i have a lot of files in the allure results, i will try to turn off the requests feature and check again. 
- macOS 13.5.1 (22G90)

**Comment by AhmadfBaker** [comment]
@mmisty Another question, Why the step of fauiler is not shown in the report like the other plugin 
Here is an example from other plugin. And in yours i just see the error in red box, but no indication from which step the code failed. 

<img width="1015" alt="image" src="https://github.com/mmisty/cypress-allure-adapter/assets/108324071/8eda8077-3e9b-4a37-b5fd-3255957d35ce">

**Reply by AhmadfBaker** [reply]
Failed Post shown in the other plugin as well. But in your plugin, no indication that this step is failed
